test/client/query-subscribe.js

```diff
@@ -592,5 +592,769 @@ module.exports = function(options) {
         });
       });
     });
+
+    describe('custom channels', function() {
+      beforeEach(function() {
+        this.backend.use('commit', function(context, next) {
+          context.channels.push('channel-1');
+          context.channels.push('channel-3');
+          next();
+        });
+        this.backend.use('query', function(context, next) {
+          context.channels = ['channel-1', 'channel-2'];
+          next();
+        });
+      });
+
+      it('should throw if not channels provided in query', function(done) {
+        this.backend.use('query', function(context, next) {
+          context.channels = null;
+          next();
+        });
+        var connection = this.backend.connect();
+        var matchAllDbQuery = this.matchAllDbQuery;
+        async.parallel([
+          function(cb) {
+            connection.get('dogs', 'fido').on('error', done).create({age: 3}, cb);
+          },
+          function(cb) {
+            connection.get('dogs', 'spot').on('error', done).create({age: 5}, cb);
+          }
+        ], function(err) {
+          if (err) return done(err);
+          connection.createSubscribeQuery('dogs', matchAllDbQuery, null, function(err) {
+            if (!err) return done('Should throw Required minimum one query channel error');
+            expect(err.message).to.be.equal('Required minimum one query channel.');
+            done();
+          });
+        });
+      });
+
+      it('should throw if channels provided in query is an empty array', function(done) {
+        this.backend.use('query', function(context, next) {
+          context.channels = [];
+          next();
+        });
+        var connection = this.backend.connect();
+        var matchAllDbQuery = this.matchAllDbQuery;
+        async.parallel([
+          function(cb) {
+            connection.get('dogs', 'fido').on('error', done).create({age: 3}, cb);
+          },
+          function(cb) {
+            connection.get('dogs', 'spot').on('error', done).create({age: 5}, cb);
+          }
+        ], function(err) {
+          if (err) return done(err);
+          connection.createSubscribeQuery('dogs', matchAllDbQuery, null, function(err) {
+            if (!err) return done('Should throw Required minimum one query channel error');
+            expect(err.message).to.be.equal('Required minimum one query channel.');
+            done();
+          });
+        });
+      });
+
+      it('creating a document updates a subscribed query', function(done) {
```

Comment:
Since many of these tests are identical to above, let's either make a shared function with the shared tests invoked for each case, or use a for-each. That reduces test maintenance.